junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/DefaultLegacyReportingInfo.java

```diff
@@ -0,0 +1,33 @@
+package org.junit.platform.engine.support.descriptor;
+
+import java.io.Serializable;
+import java.util.Optional;
+import org.junit.platform.engine.TestDescriptor;
+
+public class DefaultLegacyReportingInfo implements TestDescriptor.LegacyReportingInfo, Serializable {
+    private static final long serialVersionUID = 2896941058543948169L;
+    private final String methodName;
+    private final String className;
+
+
+    public static DefaultLegacyReportingInfo from(TestDescriptor.LegacyReportingInfo info) {
+        final String methodName = info.getMethodName().orElse(null);
+        final String className = info.getClassName().orElse(null);
```

Comment:
I feel a bit iffy about having `final` keywords on the two lines.

IMO, they feel unnecessary as the method is so short that it's reasonably easy to tell that they're never modified, and because of this, I think they only make the code a bit harder to read and maintain.